ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/FeatureProjectFactory.java

```diff
@@ -140,7 +140,15 @@ Document dom(String relative) {
         }
 
         final boolean hasFile(String relative) {
-            return dir.getFileObject(relative) != null;
+            FileObject d = dir;
+            int pos = 0;
+
+            while (relative.startsWith("../", pos) && d != null) {
```

Comment:
This is a fix that could go into `AbstractFileObject` and `masterfs` own `BaseFileObj`.

Reply:
Yes. I think there's something like this in openide.filesystems, but not in masterfs. The reason I shied away from doing this in masterfs is that this actually issues the FileObjects, and could be slower in some cases (on Linux, AFAIK, normalizeFile is very cheap). I think here, the slowdown is probably OK, but in masterfs, not so sure. I tried to implement without issuing the FOs, but that seems to be beyond my knowledge.